locallib.php

```diff
@@ -749,38 +762,55 @@ function mod_studentquiz_community_stats($cmid) {
-    $joins = mod_studentquiz_helper_attempt_stat_joins();
-    $rs = $DB->get_record_sql($select.$joins,
-        array('cmid1' => $cmid, 'cmid2' => $cmid, 'cmid3' => $cmid,
-            'cmid4' => $cmid, 'cmid5' => $cmid, 'cmid6' => $cmid, 'cmid7' => $cmid
-        ));
+    $attemptstastjoins = mod_studentquiz_helper_attempt_stat_joins($cmid, $groupid);
+    $params = [
+        'cmid1' => $cmid,
+        'cmid2' => $cmid,
+        'cmid3' => $cmid,
+        'cmid4' => $cmid,
+        'cmid5' => $cmid,
+        'cmid6' => $cmid,
+        'cmid7' => $cmid
+    ];
+    $params += $attemptstastjoins->params;
+    $rs = $DB->get_record_sql("$select {$attemptstastjoins->joins} {$attemptstastjoins->wheres}", $params);
     return $rs;
 }
 
 /**
  * Get aggregated studentquiz data
+ *
  * @param int $cmid Course module id of the StudentQuiz considered.
+ * @param int $groupid Group id.
  * @param stdClass $quantifiers ad-hoc class containing quantifiers for weighted points score.
- * @param int $userid
+ * @param int $userid User id.
  * @return array of user ranking stats
  * TODO: use mod_studentquiz_report_record type
+ * @throws dml_exception
  */
-function mod_studentquiz_user_stats($cmid, $quantifiers, $userid) {
+function mod_studentquiz_user_stats($cmid, $groupid, $quantifiers, $userid) {
     global $DB;
     $select = mod_studentquiz_helper_attempt_stat_select();
-    $joins = mod_studentquiz_helper_attempt_stat_joins();
+    $attemptstastjoins = mod_studentquiz_helper_attempt_stat_joins($cmid, $groupid);
     $addwhere = ' AND u.id = :userid ';
     $statsbycat = ' ) statspercategory GROUP BY userid';
-    $rs = $DB->get_record_sql($select.$joins.$addwhere.$statsbycat,
-        array('cmid1' => $cmid, 'cmid2' => $cmid, 'cmid3' => $cmid,
-            'cmid4' => $cmid, 'cmid5' => $cmid, 'cmid6' => $cmid, 'cmid7' => $cmid
-        , 'questionquantifier' => $quantifiers->question
-        , 'approvedquantifier' => $quantifiers->approved
-        , 'ratequantifier' => $quantifiers->rate
-        , 'correctanswerquantifier' => $quantifiers->correctanswer
-        , 'incorrectanswerquantifier' => $quantifiers->incorrectanswer
-            , 'userid' => $userid
-        ));
+    $params = [
+        'cmid1' => $cmid,
+        'cmid2' => $cmid,
+        'cmid3' => $cmid,
+        'cmid4' => $cmid,
+        'cmid5' => $cmid,
+        'cmid6' => $cmid,
+        'cmid7' => $cmid,
+        'questionquantifier' => $quantifiers->question,
+        'approvedquantifier' => $quantifiers->approved,
+        'ratequantifier' => $quantifiers->rate,
+        'correctanswerquantifier' => $quantifiers->correctanswer,
+        'incorrectanswerquantifier' => $quantifiers->incorrectanswer,
+        'userid' => $userid
+    ];
+    $params += $attemptstastjoins->params;
+    $rs = $DB->get_record_sql("$select {$attemptstastjoins->joins} {$attemptstastjoins->wheres} $addwhere $statsbycat ", $params);
```

Comment:
Has anything actually changed here?

If you want to re-format lots of code as part of a change, it is much better to have two separate comments:

1) One commit which fixes the coding style, but does not change anything else.
2) One change to implement the new feature.

That separation makes things much easier for the reviewing. (Probably too late to do anything about this now.)